base/missing.jl

```diff
@@ -69,6 +69,7 @@ convert(::Type{T}, x::T) where {T>:Union{Missing, Nothing}} = x
 convert(::Type{T}, x) where {T>:Missing} = convert(nonmissingtype_checked(T), x)
 convert(::Type{T}, x) where {T>:Union{Missing, Nothing}} = convert(nonmissingtype_checked(nonnothingtype_checked(T)), x)
 
+index_and_iterate(::Missing) = throw(MethodError(iterate, (missing,)))
```

Comment:
I assume this is there for the same reason as the one for `nothing`, but would it make sense to add a comment here as well?